src/mlpack/methods/lsh/lsh_search_impl.hpp

```diff
@@ -733,7 +762,8 @@ void LSHSearch<SortPolicy>::ReturnIndicesFromTable(
 
     // Retrieve candidates.
     size_t start = 0;
-    for (size_t i = 0; i < numTablesToSearch; ++i) // For all tables
+
+    for (long long int i = 0; i < numTablesToSearch; ++i) // For all tables
```

Comment:
The restriction to `long long` here to support Windows really irks me, and as far as I can tell there is no reasonable and portable way to get something that's the same size as `size_t` but signed (there is `ssize_t` but its support on Windows is unclear to me).  I have half a mind to just require OpenMP 3.0 support to get rid of this stupid restriction, which would disable OpenMP support with Visual Studio.  I am not sure I am bothered by that; OpenMP 3 is almost a decade old at this point and the Visual Studio team still doesn't have support for it, so I am not sure I want to keep restricting us to such a legacy version.  Windows users, if they need parallelism, can always switch to using MinGW or ICC or something like that.  What do you think?